packages/story-editor/src/components/canvas/mediaRecordingLayer.js

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@googleforcreators/i18n';
+import {
+  useCallback,
+  useEffect,
+  useRef,
+  useDebouncedCallback,
+} from '@googleforcreators/react';
+import {
+  blobToFile,
+  createBlob,
+  getImageFromVideo,
+} from '@googleforcreators/media';
+import { format } from '@googleforcreators/date';
+
+/**
+ * Internal dependencies
+ */
+import {
+  useMediaRecording,
+  PermissionsDialog,
+  ErrorDialog,
+  Countdown,
+  VideoMode,
+  ProgressBar,
+  DurationIndicator,
+  Footer,
+  SettingsModal,
+  PlayPauseButton,
+  LayerWithGrayout,
+  DisplayPageArea,
+  Wrapper,
+  PHOTO_MIME_TYPE,
+  PHOTO_FILE_TYPE,
+  VideoWrapper,
+  Video,
+  Photo,
+} from '../mediaRecording';
+import { PageTitleArea } from './layout';
+
+function MediaRecordingLayer() {
+  const {
+    updateMediaDevices,
+    status,
+    mediaBlob,
+    mediaBlobUrl,
+    setMediaBlobUrl,
+    liveStream,
+    getMediaStream,
+    setFile,
+    needsPermissions,
+    hasAudio,
+    isGif,
+    toggleIsGif,
+    isImageCapture,
+    audioInput,
+    videoInput,
+    resetStream,
+  } = useMediaRecording(({ state, actions }) => ({
+    status: state.status,
+    liveStream: state.liveStream,
+    mediaBlob: state.mediaBlob,
+    mediaBlobUrl: state.mediaBlobUrl,
+    hasAudio: state.hasAudio,
+    isGif: state.isGif,
+    needsPermissions:
+      ('idle' === state.status || 'acquiring_media' === state.status) &&
+      !state.videoInput,
+    audioInput: state.audioInput,
+    videoInput: state.videoInput,
+    isImageCapture: Boolean(state.file?.type?.startsWith('image')),
+    toggleIsGif: actions.toggleIsGif,
+    updateMediaDevices: actions.updateMediaDevices,
+    getMediaStream: actions.getMediaStream,
+    clearMediaStream: actions.clearMediaStream,
+    setFile: actions.setFile,
+    setMediaBlobUrl: actions.setMediaBlobUrl,
+    resetStream: actions.resetStream,
+  }));
+
+  const streamRef = useRef();
+  const videoRef = useRef();
+
+  const isReady = 'ready' === status;
+  const isFailed = 'failed' === status;
+
+  const isMuted = !hasAudio || isGif;
+
+  const captureImage = useCallback(async () => {
+    if (!streamRef.current) {
+      return;
+    }
+
+    const blob = await getImageFromVideo(streamRef.current);
+    try {
```

Comment:
```suggestion
    let blob;
    try {
       blob = await getImageFromVideo(streamRef.current);
```

Why not wrap this in try catch?

Reply:
I'll need to think more about the error handling there in general, but ya probably needed

Reply:
We could just wrap everything in a try catch maybe. I don't see why it not, error handling is good.

Reply:
There is more to that than just wrapping in a `try catch`. I am also thinking about user messaging etc. I will revisit this section.